src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -67,6 +67,15 @@ public virtual void AddCompilationRoots()
 
                 AddCompilationRootsForExports(module);
             }
+
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
+                if (_typeSystemContext.InputFilePaths.ContainsKey(inputFile.Value)) continue;
```

Comment:
Because I thought it caused the issue you described above with the duplicate-module exception. If it turns out it isn’t needed, I’ll remove it.